### PR TITLE
Introduce URL variables

### PR DESCRIPTION
**Pull request by TingALin**
This links to issue #531

### CODE REVIEW

File: public/favicons/polkadot/browserconfig.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<browserconfig>
+    <msapplication>
+        <tile>
+            <square150x150logo src="/mstile-150x150.png"/>
+            <TileColor>#da532c</TileColor>
```
Comment (rossbulat) [inline public/favicons/polkadot/browserconfig.xml:6]: Why is this orange?

File: public/index.html [hunk 2]
```diff
@@ -5,10 +5,10 @@
...
-  <link rel="apple-touch-icon" sizes="180x180" href="%PUBLIC_URL%/favicons/apple-touch-icon.png">
-  <link rel="icon" type="image/png" sizes="32x32" href="%PUBLIC_URL%/favicons/favicon-32x32.png">
-  <link rel="icon" type="image/png" sizes="16x16" href="%PUBLIC_URL%/favicons/favicon-16x16.png">
-  <link rel="mask-icon" href="%PUBLIC_URL%/favicons/safari-pinned-tab.svg" color="#e6007a">
+  <link rel="apple-touch-icon" sizes="180x180" href="%PUBLIC_URL%/favicons/polkadot/apple-touch-icon.png">
+  <link rel="icon" type="image/png" sizes="32x32" href="%PUBLIC_URL%/favicons/polkadot/favicon-32x32.png">
+  <link rel="icon" type="image/png" sizes="16x16" href="%PUBLIC_URL%/favicons/polkadot/favicon-16x16.png">
+  <link rel="mask-icon" href="%PUBLIC_URL%/favicons/polkadot/safari-pinned-tab.svg" color="#e6007a">
   <meta name="msapplication-TileColor" content="#e6007a">
   <meta name="theme-color" content="#fff">
```
Comment (rossbulat) [inline public/index.html:13]: I don't think these 2 meta tags are updated when the network is switched?

File: src/Router.tsx [hunk 3]
```diff
@@ -40,6 +42,68 @@ export const RouterInner = () => {
   const { sideMenuOpen, sideMenuMinimised, setContainerRefs } = useUi();
   const { t } = useTranslation('base');
 
+  const [urlNetworkExists, setUrlNetworkExists] = useState<boolean>(false);
+  const [urlNetwork, setUrlNetwork] = useState<any>('');
+  const [urlLngExists, setUrlLngExists] = useState<boolean>(false);
+  const [urlLng, setUrlLng] = useState<any>('');
```
Comment (rossbulat) [inline src/Router.tsx:48]: None of these state items are needed. The active `network` and `switchNetwork` function are stored in the `API` context, and the active locale is  managed by `i18n.resolvedLanguage` / `i18n.changeLanguage()`. 

Use these directly. If the URL vars do not exist, nothing needs to be done.

File: src/Router.tsx [hunk 4]
```diff
@@ -40,6 +42,68 @@ export const RouterInner = () => {
...
+      (urlNetworkSource === 'polkadot' || 'kusama' || 'westend')
+    ) {
+      setUrlNetworkExists(true);
+      setUrlNetwork(urlNetworkSource);
+    }
+    const _network = urlNetworkExists
+      ? urlNetwork
+      : network
+      ? network.name.toLowerCase()
+      : DefaultNetwork;
```
Comment (rossbulat) [inline src/Router.tsx:64]: Something to be careful of is that `DefaultNetwork` is all lower case, whereas `NETWORKS[x].name is capitalised. We should resolve this and stick to one convention.

**Review by rossbulat — CHANGES_REQUESTED** [review]
A start, but lots of tidy up to do. Will do some testing after the comments are addressed.

File: public/favicons/polkadot/browserconfig.xml (hunk 1 above)
Comment (TingALin) [reply public/favicons/polkadot/browserconfig.xml:6]: Just realized this is the suggested color for the Windows Metro UI and I didn't edit it,  now I know that 3 favicons are using the same tilecolor.
And for the kusama and westend icon, I chose the inline ones from the img folder, so let me know if i should use other's images for the favicons.

File: public/index.html (hunk 2 above)
Comment (TingALin) [reply public/index.html:13]: I didn't see there's any obvious differences between the background color change even they are using different color number, I will edit it

File: src/Router.tsx [hunk 5]
```diff
@@ -35,10 +40,27 @@ import {
...
+    const urlNetworkSource = extractUrlValue('n');
+    if (
+      Object.values(NETWORKS).find(
+        (n: any) => n.name.toLowerCase() === urlNetworkSource
+      )
+    ) {
+      switchNetwork(urlNetworkSource as NetworkName, true);
+    }
+
+    const urlLngSource = extractUrlValue('l');
```
Comment (rossbulat) [inline src/Router.tsx:58]: `lngFromUrl`

**Comment by rossbulat** [comment]
Please also use `t` to translate the `TitleDefault`. `TitleDefault` can be discontinued.

Screenshot 2023-01-07 at 22 52 49

File: src/library/Hooks/useUrlVars/index.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import { changeLanguage } from 'locale/utils';
+import { useTranslation } from 'react-i18next';
+import { NetworkName } from 'types';
+import { extractUrlValue } from 'Utils';
+
+export const useUrlVars = () => {
+  const { i18n } = useTranslation();
+  // get url variables
+  const networkFromUrl = extractUrlValue('n');
+  const lngFromUrl = extractUrlValue('l');
```
Comment (rossbulat) [inline src/library/Hooks/useUrlVars/index.tsx:16]: These can be inside `initialise()` at the top. They are a part of `initialise`- although there is no harm in calling them here (app will still work) let's keep all initialisation logic together.

**Comment by rossbulat** [comment]
@TingALin Can you amend the url when a network is switched? e.g. if `n=kusama` is in the URL, and a user switches to polkadot, change the URL to be `n=polkadot`.

**Pull request merged**